src/pages/resultsView/cancerSummary/CancerSummaryChart.tsx

```diff
@@ -620,6 +626,41 @@ export class CancerSummaryChart extends React.Component<
         });
     }
 
+    @autobind
+    private getData(dataType: DataType) {
+        let flatdata = this.convertDataToDownloadData(this.props.data);
+        return tsvFormatRows(
+            [
+                [this.props.xAxisString, this.yAxisLabel, 'Alteration Type'],
+            ].concat(flatdata)
+        );
+    }
+
+    @autobind
+    private convertDataToDownloadData(
+        data: {
+            x: string;
+            y: number;
+            alterationType: string;
+        }[][]
+    ): string[][] {
+        let downloadDataArray = [];
+        let rowcount = data.length;
+        let colcount = data[0].length;
+        for (var j = 0; j < colcount; j++) {
```

Comment:
you don't have to change this but in general i advocate using forEach instead of for loop.  For loop might be more performant but it's negligible.  forEach is more readable because it gives you element.

Reply:
used foreach in the earlier version, but since @jjgao wanted  the data to be sorted by x-axis labels, changed the looping to first by column and then by row.